### PR TITLE
Granular democracy

### PR DESCRIPTION
**Pull request by michaelleerilee**
Enabling user-level granule readers/plugins is a necessary idea.

Hope it doesn't break anything.

### CODE REVIEW

**Comment by NiklasPhabian** [comment]
we need to make sure it passes the tests; you can run the tests by invoking 

```bash 
pytest 
```

from the project root.

**Comment by NiklasPhabian** [comment]
alright. Looking good! Lets pull this into master. 
I will create a new release as well. FYI, we merely do this by adding a tag starting with a v. This will trigger the build action which will upload the new version to pypi

**Comment by NiklasPhabian** [comment]
one more thing to consider doing is making sure the documentation builds and is satisfactory.
you can build the documentation locally by going to the docs folder and running the makefile.

```bash
cd docs/
make html
```

File: starepandas/tools/trixel_conversions.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 import pystare
 import shapely
 import geopandas
-
+from shapely.geometry import Point
```
Comment (NiklasPhabian) [inline starepandas/tools/trixel_conversions.py:9]: seems common practice, but I am personally not a huge fan!

Comment (michaelleerilee) [reply starepandas/tools/trixel_conversions.py:9]: feel free to modify

File: starepandas/io/granules/granule.py [hunk 2]
```diff
@@ -40,6 +40,8 @@ def guess_companion_path(self, prefix=None, folder=None):
             prefix = self.companion_prefix
         return starepandas.io.granules.guess_companion_path(self.file_path, prefix=prefix, folder=folder)
 
+#    def add_stare(self, adapt_resolution=True):
+#        self.stare = pystare.from_latlon_2d(lat=self.lat, lon=self.lon, adapt_level=adapt_resolution)
     def add_sids(self, adapt_resolution=True):
```
Comment (NiklasPhabian) [inline starepandas/io/granules/granule.py:45]: ah; right. We renamed 'stare' to 'sids' to distinguish from 'tids'

File: starepandas/tools/trixel_conversions.py (hunk 1 above)
Comment (michaelleerilee) [reply starepandas/tools/trixel_conversions.py:9]: I can do this

**Review by NiklasPhabian — APPROVED** [review]

File: starepandas/tools/trixel_conversions.py (hunk 1 above)
Comment (michaelleerilee) [reply starepandas/tools/trixel_conversions.py:9]: nvm

Comment (NiklasPhabian) [reply starepandas/tools/trixel_conversions.py:9]: hahaha. Yeah; I was more testing out githubs reviewing tool than anything else.

**Pull request merged**